packages/ssi-sdk-core/src/loggers/eventLogger/EventLoggerBuilder.ts

```diff
@@ -0,0 +1,27 @@
+import { IAgentContext } from '@veramo/core'
+import EventLogger from './EventLogger'
+
+class EventLoggerBuilder {
+  private context?: IAgentContext<any>;
+  private namespace?: string
+
+  withContext(context: IAgentContext<any>): this {
+    this.context = context;
+    return this;
+  }
+
+  withNamespace(namespace: string): this {
+    this.namespace = namespace;
+    return this;
+  }
+
+  public build(): EventLogger {
```

Comment:
I would also at least expect system and subsystem here, given you will want to be able to configure the event logger in a file for the same system/subsystem in a lot of cases, meaning you do not have to provide that data when doing the actual logging.

Similar I would expect a default logging level, which would be used if not provided during the actual logging.

Reply:
i added options for system, subSystemType, and logLevel. this now means that every event, regardless of context, should have these.